hs/src/Reach/AST/SL.hs

```diff
@@ -182,12 +182,46 @@ instance Equiv Char where
 instance Equiv Int where
   equiv = (==)
 
+instance Equiv T.Text where
+  equiv = (==)
+
 instance Equiv Integer where
   equiv = (==)
 
 instance Equiv Bool where
   equiv = (==)
 
+instance (Equiv a, Equiv b) => Equiv (Either a b) where
+  equiv x y = case (x,y) of
+    (Left a, Left b) -> equiv a b
+    (Right a, Right b) -> equiv a b
+    _ -> False
+
+instance Equiv PrimOp where
+  equiv = (==)
+
+instance Equiv SecurityLevel where
+  equiv = (==)
+
+instance Equiv FluidVar where
+  equiv = (==)
+
+instance Equiv DLArg where
+  equiv a b = case (a,b) of
+    (DLA_Var v1, DLA_Var v2) -> equiv v1 v2
+    (DLA_Constant c1, DLA_Constant c2) -> equiv c1 c2
+    (DLA_Literal lit1, DLA_Literal lit2) -> equiv lit1 lit2
+    (DLA_Interact part1 s1 t1, DLA_Interact part2 s2 t2) -> equiv part1 part2 && equiv s1 s2 && equiv t1 t2
```

Comment:
You can skip comparing `t1` and `t2` because they're invariant for the same `part/s`